### PR TITLE
Document OTA API

### PR DESCRIPTION
**Pull request by armandas**
I've been recently implementing OTA for my project and noticed the documentation was lacking. I decided to add documentation to improve my own understanding and hopefully make it more accessible to others.

I used [ESP-IDF OTA documentation](https://docs.espressif.com/projects/esp-idf/en/stable/esp32/api-reference/system/ota.html) for reference.

As I'm not an expert here, I would appreciate if someone could check the correctness, especially the module doc.

Additionally, I'm not sure about the contents of `EspOtaUpdate::flush()`. On one hand, this function does nothing and the user may want to know about it, on the other hand, I just exposed implementation details. Should the documentation be written with the assumption that this function does something? This would encourage users to use it and ensure compatibility with future updates.

### CODE REVIEW

File: src/ota.rs [hunk 1]
```diff
@@ -183,8 +238,8 @@ impl<'a> EspOtaUpdate<'a> {
         Ok(())
     }
 
+    /// No-op: the OTA update is aborted when `EspOtaUpdate` is dropped.
```
Comment (DaneSlattery) [inline src/ota.rs:241]: This consumes `self` without borrowing, forcing `EspOtaUpdate` to be dropped when the function is called.

The comment might be better as `Cancel the update`

File: src/ota.rs [hunk 2]
```diff
@@ -218,6 +273,7 @@ pub struct EspOtaUpdateFinished<'a> {
 }
 
 impl EspOtaUpdateFinished<'_> {
+    /// Sets the boot partition to the newly updated app partition.
```
Comment (DaneSlattery) [inline src/ota.rs:276]: Note that the user must still reboot for the updated app partition to go into effect

File: src/ota.rs [hunk 3]
```diff
@@ -239,6 +300,11 @@ impl EspOta {
         Ok(Self(()))
     }
 
+    /// Returns the currently configured boot slot.
```
Comment (DaneSlattery) [inline src/ota.rs:303]: boot slot that will go into effect on the next reboot.

**Review by DaneSlattery — COMMENTED** [review]
This is all I could think of, looks good.

**Comment by DaneSlattery** [comment]
This is for your perusal as well in case you wanted a more in depth example https://github.com/DaneSlattery/esp-ota

File: src/ota.rs (hunk 3 above)
Comment (armandas) [reply src/ota.rs:303]: Thanks. I feel like this is already implied, i.e. "boot slot" will obviously be used for the next boot. Or am I missing something?

File: src/ota.rs (hunk 2 above)
Comment (armandas) [reply src/ota.rs:277]: I added "The app will be run on the next boot.". Does that sound OK?

File: src/ota.rs (hunk 3 above)
Comment (DaneSlattery) [reply src/ota.rs:305]: No I think you're right

**Comment by ivmarkov** [comment]
Thanks!

**Pull request merged**